packages/puppeteer-core/src/node/ChromeLauncher.ts

```diff
@@ -93,31 +82,23 @@ export class ChromeLauncher implements ProductLauncher {
     if (userDataDirIndex < 0) {
       isTempUserDataDir = true;
       chromeArguments.push(
-        `--user-data-dir=${await fs.promises.mkdtemp(
-          path.join(tmpdir(), 'puppeteer_dev_chrome_profile-')
-        )}`
+        `--user-data-dir=${await mkdtemp(this.getProfilePath())}`
       );
       userDataDirIndex = chromeArguments.length - 1;
     }
 
     const userDataDir = chromeArguments[userDataDirIndex]!.split('=', 2)[1];
     assert(typeof userDataDir === 'string', '`--user-data-dir` is malformed');
 
-    let chromeExecutable = executablePath;
-    if (channel) {
-      // executablePath is detected by channel, so it should not be specified by user.
+    let chromeExecutable: string;
+    if (this.puppeteer._isPuppeteerCore || executablePath) {
       assert(
-        !chromeExecutable,
-        '`executablePath` must not be specified when `channel` is given.'
+        executablePath,
+        `An \`executablePath\` must be specified for \`puppeteer-core\``
```

Comment:
is it a breaking change? it should be possible to launch puppeteer-core with a chanel only.

Reply:
It's a breaking changes It's not possible to in puppeteer core because we don't have a default path to download there anymore.

Reply:
why not keep the path mapping there? I don't think it deserves a breaking change.